[docs] Batch18 most functions in other (#9)
edgar2020/druid: docs/querying/sql-functions.md

## CASE

> Returns a result based on given conditions.

Reviewer: ```suggestion
Returns a result based on given conditions:
* Simple CASE compares an expression to a set of values or expressions.
* Searched CASE evaluates a set of Boolean expressions.
```

* **Syntax:** `CASE expr WHEN value1 THEN result1 \[ WHEN value2 THEN result2 ... \] \[ ELSE resultN \] END`
* **Syntax:** `CASE WHEN boolean_expr1 THEN result1 \[ WHEN boolean_expr2 THEN result2 ... \] \[ ELSE resultN \] END`
* **Function type:** Scalar, other

<details><summary>Examples</summary>

> The following example returns a string representing the UI type based on the value of `agent_category` from the `kttm` datasource.

Reviewer: ```suggestion
The following example returns the UI type corresponding to the value of `agent_category` from the `kttm` datasource.
```

```sql
SELECT "agent_category" AS "device_type",
CASE "agent_category"
    WHEN 'Personal computer' THEN 'Large UI'
    WHEN 'Smartphone' THEN 'Mobile UI'
    ELSE 'other'
END AS "UI_type"
FROM "kttm"
LIMIT 2
```

Returns the following:

| `device_type` | `UI_type` |
| -- | -- |
| `Personal computer` | `Large UI` |
| `Smartphone` | `Mobile UI` |

> The following example returns a string based on the value of the `OriginStateName` column from the `flight-carriers` datasource.

Reviewer: ```suggestion
The following example returns the departure location corresponding to the value of the `OriginStateName` column from the `flight-carriers` datasource.
```

```sql
SELECT "OriginStateName" AS "flight_origin",
CASE
    WHEN "OriginStateName" = 'Puerto Rico' THEN 'U.S. Territory'
    WHEN "OriginStateName" = 'U.S. Virgin Islands' THEN 'U.S. Territory'
    ELSE 'U.S. State'
END AS "state_status"
FROM "flight-carriers"
LIMIT 2
```

Returns the following:

| `flight_origin` | `departure_location` |
| -- | -- |
| `Puerto Rico` | `U.S. Territory` |
| `Massachusetts` | `U.S. State` |

</details>

[Lean more](sql-scalar.md#other-scalar-functions)

## CAST

Converts a value into the specified data type.

* **Syntax:** `CAST(value AS TYPE)`
* **Function type:** Scalar, other

<details><summary>Example</summary>

> The following example converts the values in the `Distance` column from the `flight-carriers` datasource from `DOUBLE` into type `VARCHAR`

Reviewer: ```suggestion
The following example converts the values in the `Distance` column from the `flight-carriers` datasource from `DOUBLE` to `VARCHAR`.
```

## COALESCE

Returns the first non-null value.
* **Syntax:** `COALESCE(expr, expr, ...)`
* **Function type:** Scalar, other

<details><summary>Example</summary>

> The following example returns the first non-null value from `null`, `null`, `5`, and `abc`.

Reviewer: ```suggestion
The following example returns the first non-null value from the list of parameters.
```

## NULLIF

Returns null if two values are equal, else returns the first value.
* **Syntax:** `NULLIF(value1, value2)`
* **Function type:** Scalar, other

<details><summary>Example</summary>

The following example returns null if the `OriginState` column from the `flight-carriers` datasource is `PR`.

```sql
SELECT "OriginState" AS "origin_state",
  NULLIF("OriginState", 'PR') AS "remove_pr"
FROM "flight-carriers"
LIMIT 2
```

Returns the following:

| `origin_state` | `remove_pr` |
| -- | -- |
| `PR` | `null` |
| `MA` | `MA` |

</details>

[Learn more](sql-scalar.md#other-scalar-functions)

## NVL

Returns `value1` if `value1` is not null, otherwise returns `value2`.

* **Syntax:** `NVL(value1, value1)`
* **Function type:** Scalar, other

<details><summary>Example</summary>

> The following example returns "No tail number" if the `Tail_Number` column from the `flight-carriers` datasource is null.

Reviewer: ```suggestion
The following example replaces each null value in the `Tail_Number` column of the `flight-carriers` datasource with the string "No tail number."
```
